snapshots/windows/windows_test.go

```diff
@@ -0,0 +1,43 @@
+//go:build windows
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package windows
+
+import (
+	"context"
+	"testing"
+
+	"github.com/containerd/containerd/pkg/testutil"
+	"github.com/containerd/containerd/snapshots"
+	"github.com/containerd/containerd/snapshots/testsuite"
+)
+
+func newSnapshotter(ctx context.Context, root string) (snapshots.Snapshotter, func() error, error) {
+	snapshotter, err := NewSnapshotter(root)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return snapshotter, func() error { return snapshotter.Close() }, nil
+}
+
+func TestWindows(t *testing.T) {
+	testutil.RequiresRoot(t)
```

Comment:
It is no-op function. Does it work?

Reply:
Huh, interesting that this is a no-op. Maybe we should implement that some day (to check for elevation though, not Administrator), and see if the test-suite can pass as non-root (and work out if anything's needlessly skipped when that's done).

That said, the Windows snapshotter absolutely relies on being run elevated (since the differ grants itself privileges, and the tests need to mount the results for validation), so this seems like the right thing to call here. The Native Snapshotter test suite makes the same check for (approximately) the same reasons.

Reply:
I think the CI is running with root role. So it can pass. Perhaps we can add check and abort if it is not root. It can be handled in the follow-up.

Reply:
Oh yes, containerd CI runs as an elevated account on Windows, that's [the default for GHA-owned VMs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#administrative-privileges), and for the Azure-hosted VMs see, e.g., [the VM creation command](https://github.com/containerd/containerd/blob/988ee8ffef1e756039fe05539be3398e70ed3e0c/.github/workflows/windows-periodic.yml#L98).

I don't think it's worth trying to change this, either. Windows CI runs both `test` and `root-test` together right now, and I don't see specific value in splitting that.

So making `RequiresRoot` skip tests when non-elevated (as the Unix version does) would simply give a better UX to developers who run the test suite on Windows from a non-elevated account, either by accident or because the individual test the user is interested in does not require 'root'. I agree _this_ is probably worth doing independently of this PR.